src/setuptools_scm/version.py

```diff
@@ -300,6 +300,13 @@ def release_branch_semver(version):
     return release_branch_semver_version(version)
 
 
+def no_guess_dev_version(version):
```

Comment:
this drops in a version that sorts formally wrong, postrelease_version implementation does this with a version that sorts formally right buut doesnt add dev tags

to be correct this should add `post1.devN` to denote a formally correct sorting of the generated version after the last tag

Reply:
Thanks for the review. I'll update.